source/ranges.tex

```diff
@@ -12882,3 +12919,652 @@
+
+\indexlibrarydtor{generator}%
+\begin{itemdecl}
+~generator();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+if (@\exposid{coroutine_}@) {
+  @\exposid{coroutine_}@.destroy();
+}
+\end{codeblock}
+
+\pnum
+\begin{note}
+Ownership of recursively yielded generators
+is held in awaitable objects
+in the coroutine frame of the yielding generator,
+so destroying the root generator
+effectively destroys the entire stack of yielded generators.
+\end{note}
+\end{itemdescr}
+
+\indexlibrarymember{operator=}{generator}%
+\begin{itemdecl}
+generator& operator=(generator other) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+swap(@\exposid{coroutine_}@, other.@\exposid{coroutine_}@);
+swap(@\exposid{active_}@, other.@\exposid{active_}@);
+\end{codeblock}
+
+\pnum
+\returns
+\tcode{*this}.
+
+\pnum
+\begin{note}
+Iterators previously obtained from \tcode{other} are not invalidated;
+they become iterators into \tcode{*this}.
+\end{note}
+\end{itemdescr}
+
+\indexlibrarymember{begin}{generator}%
+\begin{itemdecl}
+@\exposid{iterator}@ begin();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\exposid{coroutine_} refers to a coroutine
+suspended at its initial suspend point\iref{dcl.fct.def.coroutine}.
+
+\pnum
+\effects
+Pushes \exposid{coroutine_} into \tcode{*\exposid{active_}},
+then evaluates \tcode{\exposid{coroutine_}.resume()}.
+
+\pnum
+\returns
+An \exposid{iterator} object
+whose member \exposid{coroutine_}
+refers to the same coroutine as does
+\exposid{coroutine_}.
+
+\pnum
+\begin{note}
+A program that calls \tcode{begin} more than once on the same generator
+has undefined behavior.
+\end{note}
+\end{itemdescr}
+
+\indexlibrarymember{end}{generator}%
+\begin{itemdecl}
+default_sentinel_t end() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{default_sentinel}.
+\end{itemdescr}
+
+\rSec2[coro.generator.promise]{class \tcode{generator::promise_type}}
+
+\begin{codeblock}
+namespace std {
+  template<class Ref, class V, class Allocator>
+  class generator<Ref, V, Allocator>::promise_type {
+  public:
+    generator get_return_object() noexcept;
+
+    suspend_always initial_suspend() const noexcept { return {}; }
+    auto final_suspend() noexcept;
+
+    suspend_always yield_value(yielded val) noexcept;
+
+    auto yield_value(const remove_reference_t<yielded>& lval)
+      requires is_rvalue_reference_v<yielded> &&
+        @\libconcept{constructible_from}@<remove_cvref_t<yielded>, const remove_reference_t<yielded>&>;
+
+    template<class R2, class V2, class Alloc2, class Unused>
+      requires @\libconcept{same_as}@<typename generator<T2, V2, Alloc2>::yielded, yielded>
+      auto yield_value(ranges::elements_of<generator<T2, V2, Alloc2>&&, Unused> g) noexcept;
+
+    template<ranges::\libconcept{input_range} Rng, class Alloc>
+      requires @\libconcept{convertible_to<}@ranges::range_reference_t<Rng>, yielded>
```

Comment:
https://github.com/cplusplus/draft/pull/5695/files#r933905143 is _not_ resolved.
```suggestion
      requires @\libconcept{convertible_to}@<ranges::range_reference_t<Rng>, yielded>
```